src/coreclr/src/jit/lower.cpp

```diff
@@ -3284,9 +3284,25 @@ GenTree* Lowering::LowerDirectCall(GenTreeCall* call)
         {
             // Non-virtual direct calls to addresses accessed by
             // a single indirection.
-            GenTree* cellAddr = AddrGen(addr);
-            GenTree* indir    = Ind(cellAddr);
-            result            = indir;
+            GenTree* cellAddr;
+#if defined(FEATURE_READYTORUN_COMPILER) && defined(TARGET_ARMARCH)
+            if (call->IsR2RRelativeIndir())
+            {
+                cellAddr = PhysReg(REG_R2R_INDIRECT_PARAM);
```

Comment:
I think it's a mistake to use a `GT_PHYSREG` here. IMO we should eliminate those, as their semantics are unclear.
I think the right answer is to return `nullptr` in this case. Then, in `BuildCall`, you would request an internal register for this case (`buildInternalIntRegisterDefForNode(call)`) as is currently done on Arm32 for the null `ctrlExpr` case. This register would then be used in codegen to load from `X11` and use as the call target.

Reply:
Yes, I chatted with @BruceForstall  yesterday and trying out https://github.com/dotnet/runtime/pull/36712. If you don't mind please have a look into that PR to make sure I captured it correctly.